### PR TITLE
Patch the node capacity with GPU memory

### PR DESCRIPTION
**Pull request by mamy-CS**
- Patch the node capacity with accelerator GPU memory custom resource for all clusters
https://github.com/openshift/instaslice-operator/issues/86

### CODE REVIEW

File: internal/controller/instaslice_daemonset.go [hunk 1]
```diff
@@ -857,6 +856,15 @@ func (r *InstaSliceDaemonsetReconciler) SetupWithManager(mgr ctrl.Manager) error
 				}
 			}
 		}
+
+		// Patch the node capacity with GPU memory in emulator mode
+		if emulatorMode == "true" {
+			totalEmulatedGPUMemory := 80
```
[the remark this answers is not in the document]
Comment (mamy-CS) [reply internal/controller/instaslice_daemonset.go]: Good enhancement, done.

[the remark this answers is not in the document]
Comment (asm582) [reply internal/controller/instaslice_daemonset.go]: Good work @mamy-CS I think we need to loop through this section and extract totalGPUmemory:

```
   MigGPUUUID:
      GPU-kind-worker10-8d042338-e67f-9c48-92b4-5b55c7e5133c: NVIDIA A100-PCIE-40GB
      GPU-kind-worker10-31cfe05c-ed13-cd17-d7aa-c63db5108c24: NVIDIA A100-PCIE-40GB
```

[the remark this answers is not in the document]
Comment (asm582) [reply internal/controller/instaslice_daemonset.go]: We need to fix the logic and add test case before getting this PR merged

[the remark this answers is not in the document]
Comment (harche) [reply internal/controller/instaslice_daemonset.go]: > We need to fix the logic and add test case before getting this PR merged

/hold

[the remark this answers is not in the document]
Comment (mamy-CS) [reply internal/controller/instaslice_daemonset.go]: @asm582 did you mean for the emulator mode? For actual GPUs, that's how `totalGPUMemory` 
 is calculated [here](https://github.com/openshift/instaslice-operator/blob/9bf141790a8140d0a60fac9bf1c2032b49231d4d/internal/controller/instaslice_daemonset.go#L900C2-L912C3)
 I will add test case for it.

**Review by sairameshv — APPROVED** [review]
/lgtm

**Comment by harche** [comment]
/hold 

@mamy-CS you may want to squash the commits.

**Comment by asm582** [comment]
Thanks for this PR, don't want to hold this longer but we certainly appreciate unit tests :) 

/lgtm

**Pull request merged**